presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+                    if (m.getValue().getRight().equals(qual)) {
+                        tryAgain = true;
+                        break;
+                    }
+                }
+
+                if (!tryAgain) {
+                    mapping.put(column.getName(), Pair.of(fam, qual));
+                }
+            }
+            while (tryAgain);
+        }
+        return mapping;
+    }
+
+    /**
+     * Creates the index tables from the given Accumulo table. No op if
+     * {@link AccumuloTable#isIndexed()} is false.
+     *
+     * @param table Table to create index tables
+     */
+    private void createIndexTables(AccumuloTable table)
+    {
+        try {
+            // If our table is indexed, create the index as well
+            if (table.isIndexed()) {
+                // Create index table and set the locality groups
+                Map<String, Set<Text>> indexGroups = Indexer.getLocalityGroups(table);
+                conn.tableOperations().create(table.getIndexTableName());
+                conn.tableOperations().setLocalityGroups(table.getIndexTableName(), indexGroups);
+
+                // Create index metrics table, attach iterators, and set locality groups
+                conn.tableOperations().create(table.getMetricsTableName());
+                conn.tableOperations().setLocalityGroups(table.getMetricsTableName(), indexGroups);
+                for (IteratorSetting s : Indexer.getMetricIterators(table)) {
+                    conn.tableOperations().attachIterator(table.getMetricsTableName(), s);
+                }
+            }
+        }
+        catch (Exception e) {
+            dropTable(table);
+            throw new PrestoException(INTERNAL_ERROR,
+                    "Accumulo error when creating index tables, check coordinator logs for more detail",
+                    e);
+        }
+    }
+
+    /**
+     * Drops the table metadata from Presto. If this table is internal, the Accumulo
+     * tables are also deleted.
+     *
+     * @param table The Accumulo table
+     */
+    public void dropTable(AccumuloTable table)
+    {
+        SchemaTableName stName = new SchemaTableName(table.getSchema(), table.getTable());
+        String tn = table.getFullTableName();
+
+        // Remove the table metadata from Presto
+        metaManager.deleteTableMetadata(stName);
+
+        // If the table is external, we won't clean up the Accumulo tables
+        if (table.isExternal()) {
+            return;
+        }
+
+        try {
+            // delete the table
+            conn.tableOperations().delete(tn);
+
+            // and delete our index tables
+            if (table.isIndexed()) {
+                conn.tableOperations().delete(Indexer.getIndexTableName(stName));
+                conn.tableOperations().delete(Indexer.getMetricsTableName(stName));
+            }
+        }
+        catch (Exception e) {
+            throw new PrestoException(INTERNAL_ERROR, "Failed to delete table",
```

Comment:
I'd consider leveraging transaction mechanism introduced in 0.132. Let `rollback` do the cleanup (of course you have to provide a rollback implementation). This could be cleaner than tons of try-catch blocks with rollback methods in place.

What to do when rollback procedure fails?
I think documenting this scenario and erroring-out with proper information (so a user can figure out how to solve the problem) is sufficient. The same way Hive connector works.